src/main/java/net/snowflake/client/core/arrow/VectorTypeConverter.java

```diff
@@ -0,0 +1,27 @@
+package net.snowflake.client.core.arrow;
+
+import net.snowflake.client.core.DataConversionContext;
+import net.snowflake.client.core.SFException;
+import net.snowflake.client.jdbc.SnowflakeType;
+import org.apache.arrow.vector.complex.FixedSizeListVector;
+
+public class VectorTypeConverter extends AbstractArrowVectorConverter {
```

Comment:
should we make our converters internal with the annotation?

Reply:
Other converters are not marked as internal api. But I marked mine as `@SnowflakeJdbcInternalApi`

Reply:
I'm not entirely sure we should. I believe that when we implement arrow batches, this may be actually helpful for customers... @sfc-gh-pmotacki wdyt?

Reply:
So far all converters were public. It make sens to give customers API to convert snowflake types. I think that your new Converter should be set the same.

Reply:
done